orchestration/templates/noop-flow.yaml

```diff
@@ -0,0 +1,52 @@
+## No-op flow for testing out argo lifecycle hooks
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: noop-flow
+spec:
+  entrypoint: main
+  onExit: {{ if .Values.notification.onlyOnFailure }}send-slack-notification-if-failed{{ else }}send-slack-notification{{ end }}
+  serviceAccountName: {{ .Values.serviceAccount.k8sName }}
+  templates:
+    - name: main
+      container:
+        image: docker/whalesay
+        command: [cowsay]
+        args: ["hello world"]
+        resources: # limit the resources
+          limits:
+            memory: 32Mi
+            cpu: 100m
+    ##
+    ## Send a Slack notifcation if the workflow has failed.
+    ##
+    - name: send-slack-notification-if-failed
+      steps:
+        - - name: send-notification
+            template: send-slack-notification
+            when: {{ "{{workflow.status}} != Succeeded" | quote }}
+    ## send a slack notification
+    ## TBD: can this be a template ref
+    - name: send-slack-notification
+      container:
+        image: curlimages/curl:7.70.0
+        env:
+          - name: WORKFLOW_ID
+            value: {{ "{{workflow.name}}" | quote }}
+          - name: WORKFLOW_STATE
+            value: {{ "{{workflow.status}}" | quote }}
+          - name: OAUTH_TOKEN
+            valueFrom:
+              secretKeyRef:
+                name: {{ .Values.notification.oauthToken.secretName }}
```

Comment:
I think this should magically thread through after [this PR](https://github.com/broadinstitute/monster-deploy/pull/102) is merged